core/src/sigverify_stage.rs

```diff
@@ -248,7 +250,16 @@ impl SigVerifyStage {
         discard_time.stop();
 
         let mut verify_batch_time = Measure::start("sigverify_batch_time");
-        let batches = verifier.verify_batches(batches, num_valid_packets);
+        let mut batches = verifier.verify_batches(batches, num_valid_packets);
+        verify_batch_time.stop();
+
+        let mut shrink_time = Measure::start("sigverify_shrink_time");
+        let num_valid_packets = count_valid_packets(&batches);
+        if num_packets / num_valid_packets > 4 {
+            shrink_batches(&mut batches);
```

Comment:
Are you missing a truncate here? I see it in test and seems like shrink_batches returns the index at which packets become discardable. Maybe the intention is just to coalesce discard/non-discard packets I don't have context.